### PR TITLE
Remove string dequoting from `KToken`

### LINKED ISSUE #140: Add tests for token encoding
Add integration tests for iteratively exercising
- `kore-rpc`
- `kprove`

interfaces with terms containing tokens of type
- `String`
- `Bytes`
- other

### LINKED ISSUE #314: `KToken` should't eagerly dequote strings
This causes a performance bottleneck for `KAst.from_dict`: https://github.com/runtimeverification/pyk/pull/312#issuecomment-1489913642

Suggested solution: delay dequoting until it is necessary, in particular, until KAST-to-KORE transformation. See: https://github.com/runtimeverification/pyk/pull/131#pullrequestreview-1211407461

Related issue: https://github.com/runtimeverification/pyk/issues/145
Related issue: https://github.com/runtimeverification/k/issues/3082

### PR DESCRIPTION
**Pull request by tothtamas28**
~Blocked on: https://github.com/runtimeverification/haskell-backend/issues/3563~ **Edit**: skipping tests

Fixes #140
Fixes #314

Removes string dequoting from `KToken`, which now occurs on KAST-to-KORE conversion.

### CODE REVIEW

**Comment by tothtamas28** [comment]
@ehildenb, some newly added integration tests fail due to https://github.com/runtimeverification/haskell-backend/issues/3563. Should I just disable them and proceed with this PR?

**Comment by ehildenb** [comment]
@tothtamas28 please also test this with KEVM Foundry integration locally, and make sure it works prior to merge. If the tests there aren't passing, then something is going wrong with the bytes encoding choices.

**Comment by ehildenb** [comment]
Also, does this have any performance impact?

**Comment by tothtamas28** [comment]
> Also, does this have any performance impact?

It makes KAST <-> JSON much faster, and KAST <-> KORE somewhat slower. Here are some measurements.

|              |   old |   new | speedup |
|--------------|-------|-------|---------|
| JSON -> KAST | 0.639 | **0.017** |     38x |
| KAST -> JSON | 0.594 | **0.002** |    297x |
| KAST -> KORE | **0.016** | 0.022 |    -30% |

File: src/tests/integration/test_string.py [hunk 1]
```diff
@@ -143,6 +143,12 @@ def config(k: Pattern | None, s: Pattern) -> Pattern:
             ),
         )
 
+    try:
+        text.encode('latin-1')
+    except ValueError:
+        # https://github.com/runtimeverification/haskell-backend/issues/3563
+        pytest.skip()
```
[the remark this answers is not in the document]
Comment (tothtamas28) [reply src/tests/integration/test_string.py]: Done: #348

**Pull request merged**